cmd/bucket-replication.go

```diff
@@ -732,7 +732,12 @@ func equals(k1 string, keys ...string) bool {
 }
 
 // returns replicationAction by comparing metadata between source and target
-func getReplicationAction(oi1 ObjectInfo, oi2 minio.ObjectInfo) replicationAction {
+func getReplicationAction(oi1 ObjectInfo, oi2 minio.ObjectInfo, opType replication.Type) replicationAction {
+	// Avoid resyncing null versions created prior to enabling replication if target has a newer copy
+	if opType == replication.ExistingObjectReplicationType &&
+		oi1.ModTime.Unix() > oi2.LastModified.Unix() && oi1.VersionID == nullVersionID {
+		return replicateNone
```

Comment:
I don't fully understand replication, but isn't better to replicate null when this object doesn't have 'REPLICA' ?

Reply:
> We should log these or some how track that they are ignored. - We can use eventType `ReplicationNotTracked` here ?

added logging

Reply:
> I don't fully understand replication, but isn't better to replicate null when this object doesn't have 'REPLICA' ?

@vadmeste , this check is needed here because `null` version is not immutable. If existing object replication is enabled and the target bucket already has a `null` version of the object that is more recent, replication process should not end up overwriting the newer content. 

btw `REPLICA` is the status on the receiving side of replication activity. If an object enters the replication queue, it has qualified some replication rule based on its state.